File: output/gemma4-e2b/avatar/Java/Python/atcoder_ABC172_D.py
import sys

def solve():
    # Read input n
    try:
        # Read the single integer n from standard input
        n = int(sys.stdin.read().strip())
    except:
        # Handle case where input might be empty
        return

    ans = 0
    # The loop runs for i from 1 to n (inclusive)
    for i in range(1, n + 1):
        # Calculate k = floor(n / i)
        k = n // i
        
        # Calculate the term: i * k * (k + 1) / 2
        # This corresponds to the Java expression: i * (n/i) * ((n/i) + 1) / 2
        # We use integer division // for the final division.
        term = i * k * (k + 1) // 2
        ans += term
        
    print(ans)

solve()

File: output/gemma4-e2b/avatar/Python/Java/atcoder_ABC172_D.java
import java.util.Scanner;

public class atcoder_ABC172_D {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read N from standard input, mimicking the behavior of the Python code reading input.
        if (scanner.hasNextInt()) {
            int N = scanner.nextInt();
            long ans = 0;
            
            // Calculate the sum based on the formula:
            // ans += i * ( N // i ) * ( N // i + 1 ) // 2
            for (int i = 1; i <= N; i++) {
                // N // i is the floor division (integer division)
                long k = N / i;
                // Calculate i * k * (k + 1) / 2
                long term = (long)i * k * (k + 1) / 2;
                ans += term;
            }
            System